### PR TITLE
- Created DAO, Service, Injector and Storage

### PR DESCRIPTION
**Pull request by NikolayFatkullin**

### CODE REVIEW

File: src/main/java/com/shop/model/Car.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+
+public class Car {
+    private Long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(Long id, String model, Manufacturer manufacturer, List<Driver> drivers) {
+        this.id = id;
```
Comment (AndreiYazadzhi) [inline src/main/java/com/shop/model/Car.java:12]: Do yo need id in constructor?

**Review by AndreiYazadzhi — APPROVED** [review]
I`m aprove this but correct comment)

File: src/main/java/com/shop/model/Car.java (hunk 1 above)
Comment (NikolayFatkullin) [reply src/main/java/com/shop/model/Car.java:12]: Oh, Thank you, didn't see that =)

File: src/main/java/com/shop/model/Car.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import java.util.List;
+
+public class Car {
+    private Long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(String model, Manufacturer manufacturer, List<Driver> drivers) {
```
Comment (Sofasmile) [inline src/main/java/com/shop/model/Car.java:11]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/03_dao_service_checklist.html
last point

File: src/main/java/com/shop/service/impl/ManufacturerServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import com.shop.lib.Inject;
+import com.shop.lib.Service;
+import com.shop.model.Manufacturer;
+import com.shop.service.ManufacturerService;
+import java.util.List;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    ManufacturerDao manufacturerDao;
```
Comment (Sofasmile) [inline src/main/java/com/shop/service/impl/ManufacturerServiceImpl.java:13]: how about access modifier?

File: src/main/java/com/shop/service/impl/ManufacturerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        manufacturerDao.create(manufacturer);
+        return manufacturer;
```
Comment (Sofasmile) [inline src/main/java/com/shop/service/impl/ManufacturerServiceImpl.java:18]:
```suggestion
        return manufacturerDao.create(manufacturer);
```

File: src/main/java/com/shop/service/impl/ManufacturerServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return manufacturerDao.getAll();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        manufacturerDao.update(manufacturer);
+        return manufacturer;
```
Comment (Sofasmile) [inline src/main/java/com/shop/service/impl/ManufacturerServiceImpl.java:34]:
```suggestion
        return manufacturerDao.update(manufacturer);
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/shop/model/Car.java [hunk 6]
```diff
@@ -8,10 +9,10 @@
     private Manufacturer manufacturer;
     private List<Driver> drivers;
 
-    public Car(String model, Manufacturer manufacturer, List<Driver> drivers) {
+    public Car(String model, Manufacturer manufacturer) {
         this.model = model;
         this.manufacturer = manufacturer;
-        this.drivers = drivers;
+        this.drivers = new ArrayList<>();
```
Comment (Sofasmile) [inline src/main/java/com/shop/model/Car.java:15]:
```suggestion
        drivers = new ArrayList<>();
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**